framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
I'm not sure I agree for multiple reasons. First from a design prespective we tend to start seeing these default arguments piling up and then people ignore them because you need to pass parameters than a pile of default arguments which you kind of don't care about because.. hey, defaults!

More importantly though we don't always need it even for a particular class. It's not static. We use this particular MeshModifier in a lot of contexts and it works fine. What's special about the test case Andy provided is that he's specifically mod'ing the mod. Unless you are doing that you don't need to delay the initialization.